### DISCUSSION TITLE
question about "expose open volumes"

### DISCUSSION THREAD

**Discussion by Sozimajor** [hardcore-sushi/DroidFS#280] [Q&A]
Hi fellow humans.
Im not sure if this is supposed to be possible, 
but lets say I have a droidfs volume containing 3 text files.
Now I enable "keep volumes open when app goes in the background" + "expose open volumes" and open the droidfs volume containing the textfiles using the volumepassword.

Should I now be able to browse my normal android file explorer and find a folder there, which contains the three text files in decrypted form?
If so, where is it located?
Its definitely not located at the path that you specify when creating a new droidfs volume, where the encrypted containers are stored.
And i wasn't able to find any of the text files by searching for the filenames in my system file explorer either.

Sorry if what im asking for is not even supposed to be a feature, 
but I thought that was the point of "expose open volumes" - 
being able to find abd open files through a normal file explorer or any other app,
while the droidfs-volume containing the files is opend.
Kind regards 
Conrad

**Comment by hardcore-sushi** [comment]
Check my previous comments [here](https://github.com/hardcore-sushi/DroidFS/issues/246#issuecomment-1777824585) and [here](https://github.com/hardcore-sushi/DroidFS/issues/246#issuecomment-1779344279).

**Reply by avineas** [reply]
If you expose open volumes, do all other apps on your device instantly get access to the contents of that volume or do they need to ask for explicit permission from the user?

Also which option do you recommend if speed isn't important, but stability: gocryptfs or cryfs. Usecase is mirroring backups to the cloud.

**Reply by hardcore-sushi** [reply]
Document providers access control is handled by the system. If your ROM doesn't deviate too much from mainline Android and has no vulnerabilities, other apps cannot access exposed volume without user interaction.

The choice between gocryptfs and cryfs is not that simple and really depends on your threat model. If you need personal recommendations, check out my [website](https://arkensys.dedyn.io), I offer online personal consulting services about digital privacy and security.

**Comment by ghost** [comment]
> If so, where is it located?

If you are using [MaterialFiles](https://github.com/zhanghai/MaterialFiles), you can only mount the folder manually. You can do this by clicking "Add storage..." on the UI. on the UI, then find your folder and select it.
